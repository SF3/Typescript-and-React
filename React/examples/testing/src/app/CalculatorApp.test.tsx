import React from "react";
import {CalculatorApp} from "./CalculatorApp";
import {render} from '@testing-library/react';

jest.mock("../calculator/Calculator", () => () => 'Calculator');
jest.mock("../history/History", () => () => 'History');

describe("Renders App Correctly", () => {
    it("Renders correct snapshot", () => {
        const target = render(
            <CalculatorApp/>
        );

        expect(target.container.firstChild).toMatchSnapshot();
    });
});
