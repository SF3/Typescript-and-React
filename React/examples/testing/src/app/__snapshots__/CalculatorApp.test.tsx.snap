// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App Correctly Renders correct snapshot 1`] = `
<div
  class="container"
>
  <h1>
    Calculator App
  </h1>
  Calculator
  <div
    class="mb-3"
  />
  History
</div>
`;
